### PR TITLE
Fix spurious zeros from jax.scipy.special.erfcx for large arguments

### LINKED ISSUE #38607: jax.scipy.special.erfcx returns spurious 0.0 in a band near x≈26.6 (should be ~0.0212; scipy/torch agree)
### Description

`jax.scipy.special.erfcx(x)` returns exactly `0.0` across a narrow band of large `x` (approximately `x ∈ [26.54, 26.64]`), where the true value is ≈ `0.0212`:

```python
import jax; jax.config.update('jax_enable_x64', True)
import jax.scipy.special as jsp, scipy.special as ssp
print(jsp.erfcx(26.6), ssp.erfcx(26.6))   # 0.0   vs 0.02119517815916648
# neighbours are correct:
print(jsp.erfcx(26.0), jsp.erfcx(27.0))    # 0.021683..., 0.020881...  (match scipy)
```

`erfcx` is strictly positive for all finite `x` (asymptotically `1/(x*sqrt(pi))` ≈ `0.02121` at `x=26.6`), so `0.0` is wrong. The band reproduces under eager, `jit`, and `vmap`. Neighboring points (26.0, 27.0, 30.0) are all correct, so this is a localized incorrect-zero band rather than a broad asymptotic error — consistent with `erfc(x)` underflowing to `0` inside a composed `exp(x**2) * erfc(x)` implementation instead of using the scaled asymptotic form.

### Cross-check

`scipy.special.erfcx(26.6)` and `torch.special.erfcx(26.6)` both return `0.02119517815916648`; the large-`x` asymptote `1/(x*sqrt(pi))` confirms it. JAX is the lone outlier.

(The gradient is affected downstream: `jax.grad(erfcx)(26.6)` returns `-2/sqrt(pi)` because the vanishing forward value zeroes the `2*x*erfcx(x)` term; this resolves once the value is fixed.)

### System info
```
jax 0.10.1, jaxlib 0.10.1, numpy 2.4.6, python 3.12.13, CPU
```

### PR DESCRIPTION
**Pull request by GodlyDonuts**
`jax.scipy.special.erfcx` returns 0 for a range of large x where the true value is nonzero (#38607):

```python
>>> import jax; jax.config.update('jax_enable_x64', True)
>>> import jax.scipy.special as jsp, scipy.special as ssp
>>> jsp.erfcx(26.6), ssp.erfcx(26.6)
(0.0, 0.02119517815916648)
```

`erfcx(x) = exp(x^2) * erfc(x)`. The code computes this product directly and only falls back to the asymptotic series once `exp(x^2)` would overflow (around x=26.64 for float64). But `erfc(x)` underflows to a subnormal a bit before that point, so in the band roughly [26.5, 26.64] for float64 (and [9.2, 9.42] for float32) the product is `finite * 0 = 0`.

The fix switches to the asymptotic series as soon as it is accurate to eps, rather than waiting for the overflow point. The first dropped term is `|c_n|/x^(2n)` with `c_n = (2n-1)!!/2^n`, so it falls below eps for `x >= (|c_n|/eps)^(1/(2n))`, about 13.7 for float64 and 6.9 for float32. That is well before the direct product breaks down, so no accuracy is lost.

Added a regression test covering both bands. `tests/lax_scipy_special_functions_test.py` passes with x64 enabled (518 tests), and erfcx matches scipy to a few ulp across [0, 40].

Fixes #38607

### CODE REVIEW

**Comment by GodlyDonuts** [comment]
Hi @jakevdp, would you be able to review this PR when you have a chance, or route it to the appropriate reviewer? It fixes `jax.scipy.special.erfcx` returning spurious zeros for large positive arguments and adds regression tests for both float32 and float64. Happy to revise the approach if needed.

**Comment by jakevdp** [comment]
This falls under #38651 – it may take a while for us to get to it.
